structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/CustomCrnOrNameProvider.java

```diff
@@ -0,0 +1,11 @@
+package com.sequenceiq.cloudbreak.structuredevent.rest;
+
+import java.util.Map;
+
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPOperationDetails;
+import com.sequenceiq.cloudbreak.structuredevent.event.rest.RestCallDetails;
+
+public interface CustomCrnOrNameProvider {
+
+    void provide(RestCallDetails restCallDetails, CDPOperationDetails operationDetails, Map<String, String> restParams, String nameField, String crnField);
```

Comment:
it would be more "clean" if this method would return with the restParams, indicating it modifies it's content.
The most elegant way would be to make a copy of the restParams parameter to initialize the result and add values to that one, as the parameter could be even an immutable map